macros/sql/get_column_values.sql

```diff
@@ -11,19 +10,45 @@ Returns:
     A list of distinct values for the specified columns
 #}
 
-{% macro get_column_values(table, column, max_records=none) -%}
+{% macro get_columns(table, column, max_records=none, fail=True, default='blank') -%}
+
+{#-- Prevent querying of db in parsing mode. This works because this macro does not create any new refs. #}
+    {%- if not execute -%}
+        {{ return('') }}
+    {% endif %}
+{#--  #}
+
+    {%- set table_value = adapter.get_relation(database=table.database,
+                                          schema=table.schema,
+                                         identifier=table.identifier) -%}
+
+    {%- call statement('get_column_values', fetch_result=true) %}
+
+        {%- if not table_value and fail -%}
+
+          {{ log("Table doesn't exist..") }}
+          {{ exceptions.raise_compiler_error("Table to pull columns from doesn't exist. Try running " ~ table ~ " and its dependencies first.") }}
```

Comment:
```suggestion
          {{ exceptions.raise_compiler_error("In get_column_values(): relation " ~ table ~ " does not exist and no default value was provided.") }}
```

I try to avoid being prescriptive in error messages unless we can be really sure about why the failure occurred. While the error message you included here is perfect if this macro is pointing to an unbuilt model, this error can also happen if:
1. a source table does not exist
2. the user made a typo when manually creating a relation

The other annoying thing to say here is that this relation might be a `view`. I tend to use the word `relation` which encompasses both tables and views, but is probably less readily understood by the majority of dbt users. Super open to keeping `table` here for clarity - call that how you see it!

Reply:
I really like your version (and have updated this to reflect that). I'm curious about `relation` vs `model` for word choice, since the docs seem to reference "your dbt models" as an inclusive term.